joblib/test/test_memmapping.py

```diff
@@ -346,10 +480,45 @@ def test_memmapping_pool_for_large_arrays(factory, tmpdir):
     finally:
         # check FS garbage upon pool termination
         p.terminate()
-        assert not os.path.exists(p._temp_folder)
+        for i in range(10):
+            sleep(.1)
+            if not os.path.exists(p._temp_folder):
+                break
+        else:
+            raise AssertionError(
+                'temporary folder of {} was not deleted'.format(p)
+            )
         del p
 
 
+@with_numpy
+@parametrize("backend", ["multiprocessing", "loky"])
+def test_parallel_crash_no_resources_leaked(backend):
+    cmd = """if 1:
+        import numpy as np
+        from joblib import Parallel, delayed
+        from testutils import crash
+
+
+        data = np.random.rand(1000)
+
+        if __name__ == "__main__":
+            Parallel(n_jobs=2, backend="{b}", max_nbytes=100)(
+                delayed(crash)(data) for i in range(1))
```

Comment:
Maybe rename `crash` to `print_filename_and_raise_valueerror` to make the test easier to follow.

Reply:
Also you could probably catch `ValueError` so this is not a real "crash" (like a segfault of the worker for instance). So the test it-self should probably be renamed. I am not sure exactly what is the purpose of this test. If a parallel function returns a value or raises an exception, in both case the parallel call will terminate successfully. So we could catch the exception in the subscript and expect the call to subprocess to have a return code of 0.

Reply:
I think that when workers error-out, their finalizers are not run (at least, I don't see the memmaps being decref'd). This simply tests that when child workers error out, and `abort_everything` is ran in the parent, we properly delete the temporary folder.

Reply:
The worker does not error out here. It sends the error to the main process which calls `abort_everything`. This function calls `shutdown(kill_workers=True)` which sends `SIGKILL` to the workers, preventing it from running the finalizer. 
I tend to agree that this function should not be called `crash` as we use this terminology for `SIGKILL` and `SIGSEGV` usually. Maybe `print_filename_and_error` to not be too long?

Reply:
Thanks for the clarification @tomMoral.

Reply:
Ah right. I think it's worth explaining this in a comment.